File: 백준/Silver/11399. ATM/ATM.py
import sys 
input = sys.stdin.readline

n = int(input())
lst = list(map(int, input().split()))

lst.sort()

ans = lst[0]

for i in range(1, n):
    lst[i] = lst[i-1]+lst[i]
    ans += lst[i]
    
print(ans)

File: 백준/Silver/11399. ATM/ATM.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] arr = br.readLine().split(" ");
        int[] peo = new int[n];
        for (int i = 0; i < n; i++) {
            peo[i] = Integer.parseInt(arr[i]);
        }
        Arrays.sort(peo);
        int ans = 0;
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += peo[i];
            ans += sum;
        }
        System.out.println(ans);
    }
}